### PR TITLE
enable/disable features in config.json

### PR DESCRIPTION
**Pull request by dbkr**
As per https://docs.google.com/document/d/1Kn-mz2dDce9Cqc4oUTl5yJXGvmTlky1_KezuwUg58x0/edit#

Replaces:
 * enableLabs setting
 * 'override' flag in labs
 * 'default' flag in labs

Un-feature-flags matrix apps since this was now overidden to be
enabled.

### CODE REVIEW

File: src/UserSettingsStore.js [hunk 1]
```diff
@@ -180,33 +190,30 @@ export default {
...
-            localStorage.getItem(`mx_labs_feature_${featureId}`) === null
-        ) {
-            return feature.default;
+
+        if (sdkConfigValue === 'enable') {
+            return true;
+        } else if (sdkConfigValue === 'enable') {
+            return false;
+        } else if (sdkConfigValue === 'labs') {
+            if (!MatrixClientPeg.get().isGuest()) {
```
Comment (lukebarnard1) [inline src/UserSettingsStore.js]: Maybe put a comment that says that guests can't access labs?

File: src/UserSettingsStore.js [hunk 2]
```diff
@@ -180,33 +190,30 @@ export default {
...
-        // Return the default if this feature has an override to be the default value or
-        // if the feature has never been toggled and is therefore not in localStorage
-        if (Object.keys(feature).includes('override') ||
-            localStorage.getItem(`mx_labs_feature_${featureId}`) === null
-        ) {
-            return feature.default;
+
+        if (sdkConfigValue === 'enable') {
+            return true;
+        } else if (sdkConfigValue === 'enable') {
```
Comment (lukebarnard1) [inline src/UserSettingsStore.js]: Probably need a negation of some kind here :wink:

File: src/UserSettingsStore.js [hunk 3]
```diff
@@ -180,33 +190,30 @@ export default {
...
+            return false;
+        } else if (sdkConfigValue === 'labs') {
+            if (!MatrixClientPeg.get().isGuest()) {
+                const userValue = localStorage.getItem(`mx_labs_feature_${featureId}`);
+                if (userValue !== null) {
```
Comment (lukebarnard1) [inline src/UserSettingsStore.js]: This is redundant

File: src/UserSettingsStore.js [hunk 4]
```diff
@@ -180,33 +190,30 @@ export default {
...
-                return f;
-            }
-        }
-        return null;
-    },
-
     isFeatureEnabled: function(featureId: string): boolean {
-        // Disable labs for guests.
-        if (MatrixClientPeg.get().isGuest()) return false;
+        const featuresConfig = SdkConfig.get()['features'];
```
Comment (lukebarnard1) [inline src/UserSettingsStore.js]: "features" doesn't strike me as particularly grep-able. How about `featuresConfig`?

Comment (dbkr) [reply src/UserSettingsStore.js]: This is what I put in the doc, so we should go back & discuss there, although featuresConfig seems a bit redundant given it's in a config file

File: src/UserSettingsStore.js (hunk 2 above)
Comment (dbkr) [reply src/UserSettingsStore.js]: oops

File: src/UserSettingsStore.js (hunk 4 above)
Comment (lukebarnard1) [reply src/UserSettingsStore.js]: `featuresDeployState`? `featuresControl`?

**Comment by lukebarnard1** [comment]
I thought the "Approve" link might let me say "LGTM apart from potential rename of 'features' "

**Pull request merged**